File: src/modules/trees/1_representation/LinkedRepresentation.ts
export class TreeNode {
    left: TreeNode | null;
    right: TreeNode | null;
    data: number;
    constructor(data: number) {
        this.data = data;
        this.left = null;
        this.right = null
    }
}

let root = new TreeNode(10);
root.left = new TreeNode(5);
root.right = new TreeNode(20);
root.left.left = new TreeNode(3);
root.left.right = new TreeNode(8);
root.right.left = new TreeNode(15);
root.right.right = new TreeNode(25);

console.log("Root Node :", root.data);

File: src/modules/trees/1_representation/LinkedRepresentation.go
package main

import "fmt"

type TreeNode struct {
	data  int
	left  *TreeNode
	right *TreeNode
}

func NewTreeNode(data int) *TreeNode {
	return &TreeNode{
		data:  data,
		left:  nil,
		right: nil,
	}
}

func main() {
	root := NewTreeNode(10)
	root.left = NewTreeNode(5)
	root.right = NewTreeNode(20)
	root.left.left = NewTreeNode(3)
	root.left.right = NewTreeNode(8)
	root.right.left = NewTreeNode(15)
	root.right.right = NewTreeNode(25)

	// Print root node data
	fmt.Println("Root Node :", root.data)
}
